edgelet/edgelet-docker/src/lib.rs

```diff
@@ -3,44 +3,19 @@
 #![deny(unused_extern_crates, warnings)]
 #![deny(clippy::all, clippy::pedantic)]
 #![allow(clippy::stutter, clippy::use_self)]
+#![deny(rust_2018_idioms)]
 
-extern crate base64;
-extern crate chrono;
-extern crate failure;
-extern crate futures;
-extern crate hyper;
 #[macro_use]
```

Comment:
`use` the individual macros instead of `#[macro_use]`ing the crate. Eg `use lazy_static::lazy_static;` and `use log::{error, info, log};`. Or for little-used macros like `lazy_static`, use the full path at the invocation, like `lazy_static::lazy_static! { ... }`

This also applies for custom derives, like `use serde_derive::Deserialize;`, or `#[derive(serde_derive::Deserialize)]`.